### PR TITLE
locale-control: add access to the locale1 dbus interface

### PR DESCRIPTION
**Pull request by sergio-costas**
This interface is required to allow to set the GDM language.

Thanks for helping us make a better snapd!
Have you signed the [license agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by kenvandine** [comment]
Link to the upstream PR https://github.com/snapcore/snapd/pull/13027

File: interfaces/builtin/locale_control.go [hunk 1]
```diff
@@ -42,6 +69,7 @@ func init() {
 		name:                  "locale-control",
 		summary:               localeControlSummary,
 		implicitOnClassic:     true,
+		implicitOnCore:        release.OnCoreDesktop,
```
Comment (kenvandine) [inline interfaces/builtin/locale_control.go:72]: @robert-ancell is this the best way to achieve this?

Comment (robert-ancell) [reply interfaces/builtin/locale_control.go:72]: I think so? systemd is provided by core, so it needs to expose this interface. The question is why this isn't already implicit on core - are you supposed to use another method for changing locale, i.e. only the system can do it?

Comment (kenvandine) [reply interfaces/builtin/locale_control.go:72]: > 

Notice the TODO that it won't work  until snappy exposes it... sounds antique

Comment (sergio-costas) [reply interfaces/builtin/locale_control.go:72]: Maybe it has to do with /etc/default/locale not being in writable...

**Review by kenvandine — APPROVED** [review]
LGTM

File: interfaces/builtin/locale_control.go (hunk 1 above)
Comment (sergio-costas) [reply interfaces/builtin/locale_control.go:72]: I commented in the upstream the changes done to make this work... let's see what they say. https://github.com/snapcore/snapd/pull/13027#issuecomment-1665373891

**Review by kenvandine — APPROVED** [review]
This looks good!

**Pull request merged**